### ISSUE TITLE
Instructions for App Registration take you to a link for registering web applications, is that correct?

### ISSUE THREAD

**Issue by ljunquera** [Azure-Samples/active-directory-b2c-ios-swift-native-msal#60]
The instructions for [App Registration](https://github.com/Azure-Samples/active-directory-b2c-ios-swift-native-msal#app-registration) say to "Follow [instructions here](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-register-applications?tabs=app-reg-ga)" which take you to a page to "Tutorial: Register a web application in Azure Active Directory B2C". If I'm creating an iOS application why is the tutorial for registering a web application. Is that correct?

The linked page says:

> A "web application" refers to a traditional web application that performs most of the application logic on the server. They may be built using frameworks like ASP.NET Core, Maven (Java), Flask (Python), and Express (Node.js).

There is also another statement:

> If you're using a native app instead (e.g. iOS, Android, mobile & desktop), learn how to register a native client application.

Which takes you to [Add a native client application to your Azure Active Directory B2C tenant](https://docs.microsoft.com/en-us/azure/active-directory-b2c/add-native-application?tabs=app-reg-ga). Do we follow the page referenced in the instructions or follow that link to the native app instructions?

When registering an application, aside from entering a Redirect URI, there is a dropdown for platform which includes:

- Public client/native (mobile & desktop)
- Web
- Single-page application (SPA)

I am trying to build a mobile app, but the instructions say "select Web" and put "https://jwt.ms" in the Redirect URI text box.

For example, in the link from the directions there is a section on [Create a client secret](https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-register-applications?tabs=app-reg-ga#create-a-client-secret) where it says:

> For a web application, you need to create an application secret. The client secret is also known as an application password. The secret will be used by your application to exchange an authorization code for an access token.

I'm not creating a web application, so this is a little confusing. I will follow them as the instructions say, but it's hard to reconcile the differences. In the [App Registration](https://github.com/Azure-Samples/active-directory-b2c-ios-swift-native-msal#app-registration) section of the GitHub instructions it says:

> Once done, you will need add the redirect URI of `msal<your-client-id-here>://auth`

but the instructions from the link said to make the Redirect URI, https://jwt.ms

It looks like that didn't work because I get this screen:
![image](https://user-images.githubusercontent.com/4500512/150787551-e2811b86-8692-4f86-885d-847e107b282d.png)

**Comment by ishtiaque23** [comment]
Recently I faced the same issue and was a bit confused by this. 

I am also confused why I need the web application registration when I am trying to integrate AD B2C authentication in the mobile. I completely understand the need of the web app later when I will make API call to my backend services. 

I found this Link most informative: https://docs.microsoft.com/en-us/azure/active-directory-b2c/configure-authentication-sample-ios-app . 
Here you will see that the whole architecture has been addressed properly. 
This specific instruction only work with the code in the "vnext" branch of this repo. Even then I failed to make a successful sign in as it says it is receiving NIL id_token. 

The code in the Master branch does not work when you try to provide the mobile app specific redirect URI. 
It simply says that "safari is unable to open the page as the URI is invalid". 

These issues need to be addressed. I can see 2 set of paths here and 2 set of supporting guideline but it is not clearly guided from the readme of this repo.
